### PR TITLE
F3273 Added notify functionality

### PR DESCRIPTION
**Pull request by MartinBelton-gov**
F3273 Added notify functionality (Gov Notify template still needs tweaking)

image

### CODE REVIEW

File: shared/FamilyHubs.Notification.Api.Contracts/MessageDto.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+namespace FamilyHubs.Notification.Api.Contracts;
+
+public record MessageDto : DtoBase<long>
+{
+    public required string RecipientEmail { get; set; }
+    public required string TemplateId { get; set; }
+    public Dictionary<string, string> TemplateTokens { get; set; } = new Dictionary<string, string>();
+    public DateTime? Created { get; set; }
+
+    public override int GetHashCode()
```
Comment (lazcool) [inline shared/FamilyHubs.Notification.Api.Contracts/MessageDto.cs:16]: GetHashCode and Equals are provided for free by using a record.

File: src/FamilyHubs.Notification.Api/Endpoints/MinimalNotifyEndPoints.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+using Microsoft.AspNetCore.Mvc;
+using Swashbuckle.AspNetCore.Annotations;
+
+namespace FamilyHubs.Notification.Api.Endpoints;
+
+public class MinimalNotifyEndPoints
+{
+    public void RegisterMinimalNotifyEndPoints(WebApplication app)
+    {
+        app.MapPost("api/notify", async ([FromBody] MessageDto request, CancellationToken cancellationToken, ISender _mediator) =>
```
Comment (lazcool) [inline src/FamilyHubs.Notification.Api/Endpoints/MinimalNotifyEndPoints.cs:15]: We should probably protect sending notifification with [Authorize] too.

File: shared/FamilyHubs.Notification.Api.Contracts/MessageDto.cs (hunk 1 above)
Comment (MartinBelton-gov) [reply shared/FamilyHubs.Notification.Api.Contracts/MessageDto.cs:16]: Yes they are, but tests such as WhenComparingTwoObjectsWithSameValuesThenReturnsTrue and WhenComparingTwoObjectsIgnorePrimaryAndForeignKeyThenReturnsTrue fail if I don't overload those methods.

File: src/FamilyHubs.Notification.Api/Endpoints/MinimalNotifyEndPoints.cs (hunk 2 above)
Comment (MartinBelton-gov) [reply src/FamilyHubs.Notification.Api/Endpoints/MinimalNotifyEndPoints.cs]: Added Authorize attribute

**Pull request merged**